py-development.Rmd

```diff
@@ -0,0 +1,540 @@
+established by the library's authors, which for numpy is `import numpy as np`
+and use functions via `np.<function_name>`. One thing to avoid is to import
+everything from a package, e.g. `from numpy import *`. If this is done with
+every package, it is almost guaranteed that the same function name will be
+available from more than one package and it will be difficult to keep track of
+if you are using the `mean` function from `numpy` or from another package that
+you also imported everything from.
+
+Subpackages and their functions can be imported via the dot syntax.
+
+```{python}
+from numpy.fft import fftfreq
+```
+
+Packages and subpackages might sound complicated to keep apart. It can be
+helpful to understand that they are folders and files in specific directory
+structure. Considering only the `numpy` packages we have mentioned so far, the
+directory structure would look like this.
+
+```
+numpy-folder/
+    script-with-the-mean-definition.py
+    fft-folder/
+        script-with-the-fftfreq-definition.py
+```
+
+### Installing packages {#py-dev-installing-packages}
+
+After Anaconda has been installed on your system, you can use the command line
+`conda` package manager or the GUI-driven `anaconda-navigator` to install
+Python packages. For comprehensive instructions on both of these, refer to the
+[official
+documentation](https://docs.continuum.io/anaconda/#navigator-or-conda). Brief
+step-by-step instructions to get up and running with `conda` follow.
+
+1. To install a new Python package from the Anaconda repositories, simply run
+   `conda install <package name>` in a terminal. You can also use the `pip`
+   package manager, but it will be easier to keep track of packages by sticking
+   to one installation method.
+2. Some packages are not available in the default Anaconda repositories. User
+   contributed packaged are available in Anaconda "channels", use `anaconda
+   search -t conda <package name>`, to find a channel with the desired package.
+   To install this package, use `conda install -c <channel name> <package
+   name>`. The [conda forge channel](https://conda-forge.github.io/) channel
+   has many of the packages not in the default repositories.
+
+## How to get help online {#py-dev-how-to-get-help-online}
+
+When reading the built-in Python help is not sufficient, there are several 
+online resources that can helpful. One of the most commonly used resources for
+data science is the Stack Exchange network wich offer Q&A sites both for
+programming related topics via [Stack Overflow][so], as well as statistics and
+machine learning via [Cross Validated][cv].
+
+If nothing relevant can be found after searching the many existing question and
+answers on these sites, it is appropriate to ask a new question! Stack Overflow
+has detailed instruction on how to create a [minimal reproducible
+example][so-mre] when asking a question to increase the chances that the
```

Comment:
In R there is the repro package to automatically create a MRE, rendered into the markdown form needed for SO. Is there something similar in Python?

Reply:
I had a brief look, but could not find anything.